module/zfs/spa.c

```diff
@@ -3751,11 +3754,40 @@ spa_l2cache_drop(spa_t *spa)
 }
 
 /*
+ * Verify encryption parameters for spa creation. If we have specified a crypt,
+ * we must have a fully speicified key, with the encryption feature enabled.
+ * Otherwise, we should not have any specified encryption parameters.
```

Comment:
I could try to do that. It would require adding some NULL checks to `dmu_objset_create_crypt_check()`. I hesitate to do that because that function happens to have been the most difficult function to get right in the entire PR. The case for pool creation is much simpler because there can't be any inheritance or clones to deal with.